entrypoint.sh

```diff
@@ -2,18 +2,25 @@
 
 set -e
 
-if [ -z "$FIREBASE_TOKEN" ]; then
-    echo "FIREBASE_TOKEN is required to run commands with the firebase cli"
-    exit 126
+if [ -z "$FIREBASE_TOKEN" ] && [ -z "$GCP_SA_KEY" ]; then
+  echo "Either FIREBASE_TOKEN or GCP_SA_KEY is required to run commands with the firebase cli"
+  exit 126
+fi
+
+if [ -n "$GCP_SA_KEY" ]; then
+  echo "Storing GCP_SA_KEY in /opt/gcp_key.json"
+  echo "$GCP_SA_KEY" | base64 -d > /opt/gcp_key.json
+  echo "Exporting GOOGLE_APPLICATION_CREDENTIALS=/opt/gcp_key.json"
+  export GOOGLE_APPLICATION_CREDENTIALS=/opt/gcp_key.json
 fi
 
 if [ -n "$PROJECT_PATH" ]; then
-    cd $PROJECT_PATH
+  cd "$PROJECT_PATH"
 fi
 
 if [ -n "$PROJECT_ID" ]; then
     echo "setting firebase project to $PROJECT_ID"
-    firebase use --add $PROJECT_ID
+    firebase use --add "$PROJECT_ID"
```

Comment:
Any specific reason you added quotes here and above?

Reply:
I use `shellcheck` in Intellij and it gives me this warning if I don't quote https://github.com/koalaman/shellcheck/wiki/SC2086

Reply:
Works for me.